diff --git a/CMakeLists.txt b/CMakeLists.txt
index d42bbb7d..3a2e4ec6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,7 +107,7 @@ if (HAVE_X11)
 endif()
 
 # Qt 5.13 deprecated QComboBox::currentIndexChanged(QString) and Qt 5.14 undid that...
-if (Qt5Widgets_VERSION VERSION_GREATER_EQUAL 5.14.0)
+if (NOT Qt5Widgets_VERSION VERSION_LESS 5.14.0)
 add_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050d00)
 endif()
 
