diff --git a/CMakeLists.txt b/CMakeLists.txt
index da34e23..87fef63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,7 +86,7 @@ else()
         PURPOSE "Needed to build audio notification support when Canberra isn't available")
     if (NOT ANDROID)
         # This is REQUIRED since you cannot tell CMake "either one of those two optional ones are required"
-        set_package_properties(Phonon4Qt5 PROPERTIES TYPE REQUIRED)
+#        set_package_properties(Phonon4Qt5 PROPERTIES TYPE REQUIRED)
     endif()
     if (Phonon4Qt5_FOUND)
         add_definitions(-DHAVE_PHONON4QT5)
