diff --git a/src/kmainwindow.cpp b/src/kmainwindow.cpp
index 20ef876..227dbf6 100644
--- a/src/kmainwindow.cpp
+++ b/src/kmainwindow.cpp
@@ -108,6 +108,7 @@ bool DockResizeListener::eventFilter(QObject *watched, QEvent *event)
     switch (event->type()) {
     case QEvent::Resize:
     case QEvent::Move:
+    case QEvent::Show:
     case QEvent::Hide:
         m_win->k_ptr->setSettingsDirty(KMainWindowPrivate::CompressCalls);
         break;
