diff --git a/komparemodellist.cpp b/komparemodellist.cpp
index dec1162..6f7c40f 100644
--- a/komparemodellist.cpp
+++ b/komparemodellist.cpp
@@ -239,10 +239,8 @@ static QString lstripSeparators(const QString& from, uint count)
     {
         return QString();
     }
-    else
-    {
-        return from.mid(position);
-    }
+
+    return from.mid(position + 1);
 }
 
 void KompareModelList::setDepthAndApplied()
@@ -1112,9 +1110,10 @@ bool KompareModelList::blendFile(DiffModel* model, const QString& fileContents)
         // Now we add the linecount difference for the hunk that follows
         int size = hunk->sourceLineCount();
 
-        for (int i = 0; i < size; ++i)
+        linesIt += size;
+        if (linesIt > lEnd)
         {
-            ++linesIt;
+            linesIt = lEnd;
         }
 
         srcLineNo += size;
